### PR TITLE
[BranchFolding][SEH] Add test to track SEH CFG optimization

### PR DESCRIPTION
**Pull request by HaohaiWen**
This test tracks BranchFolding pass which removes fall through jump and
leaves landing-pad to be machine basic block of no predecessors. It
would raise bug as introduced in #77441.

### CODE REVIEW

File: llvm/test/CodeGen/X86/branchfolding-landingpad-cfg.mir [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py UTC_ARGS: --version 4
+# RUN: llc -mtriple=x86_64-pc-windows-msvc -run-pass=branch-folder -o - %s | FileCheck %s
+---
```
Comment (phoebewang) [inline llvm/test/CodeGen/X86/branchfolding-landingpad-cfg.mir:3]: Should attach IR in the test? Besides, without IR, how can we enable async exception?

Comment (HaohaiWen) [reply llvm/test/CodeGen/X86/branchfolding-landingpad-cfg.mir:3]: This test is just for branch-folder. I hacked the entire MIR just like llvm/test/CodeGen/X86/branchfolding-ehpad.mir. We don't care what IR looks like.

Comment (HaohaiWen) [reply llvm/test/CodeGen/X86/branchfolding-landingpad-cfg.mir:3]: See fix patch https://github.com/llvm/llvm-project/pull/77608

**Review by phoebewang — APPROVED** [review]
LGTM.

**Pull request merged**